### ISSUE TITLE
Rebuilt app is showing "Installieren Sie bitte diese App aus dem Google Play Store"

### ISSUE THREAD

**Issue by deival1980** [user987654321resu/Libre2-patched-App#3]
All the rebuild steps where working properly I selected the patch option 2 with Git. Thanks for that!

Short before there is sometimes a Messagebox shown wich is meaning something about "Lizenzprüfung". Do I have to have installed the app from Google Play Store before updating with the rebuilt version?

**Comment by deival1980** [comment]
You really have to install the original app from play store before you uninstall the app again and install the rebuilt version. I have now tried that on a phone where I have a Google Play Store. You should mention that in your description.
Now I'm just missing a Libre 2 sensor for a first try.

**Comment by user987654321resu** [comment]
Maybe your patch failed. Saw this problem once with windows and patch tool

**Comment by deival1980** [comment]
Do I understand it correct that the licensing should be skipped with the rebuilt version?
I will try to see this evening if the patching worked, but as I already wrote I used the git patching option.
After installing the original app from Play Store and uninstallig it agin I could install also the rebuilt version...
Another basic question as I have no Libre 2 sensor up to now:
Does this rebuilt version of LibreLink receive glucose values via Bluetooth and forward that values to XDRIP or does it only forward the NFC scanned glucose values?

**Comment by user987654321resu** [comment]
Only values received by Bluetooth are forwarded

**Comment by acrobatba** [comment]
Hi. Im first time here. I find out about this patch on FB. Im not so much familiar with all this stuff (apk ...), but I would like to try. Just, you say in the end "Wait for more informations about which Version/Patches you need for xDrip to get the Data to Display!". You guys installed this and its working? Nothing to wait?

**Comment by elbingobonito** [comment]
> Hi. Im first time here. I find out about this patch on FB. Im not so much familiar with all this stuff (apk ...), but I would like to try. Just, you say in the end "Wait for more informations about which Version/Patches you need for xDrip to get the Data to Display!". You guys installed this and its working? Nothing to wait?

The xdrip+ app was updated recently - if you install the latest version of xdrip+ you should be able to use "libre 2 (patched)" as a source.

**Comment by sarafanyuk** [comment]
Everything works, thanks!!! [https://drive.google.com/open?id=1_kGc_LiqfK3hAJeXV6Y66vGuD9f1XcYg](url)

**Comment by user987654321resu** [comment]
do not paste links to the patched app here!

**Comment by Rig22** [comment]
Erst einmal "Vielen Dank" für Deine Arbeit. :) Mein Pocophone F1 hat kein NFC und LibreLink verweigert aus dem Grund den Start. Eigentlich brauche ich NFC ja gar nicht, da ich per Bluetooth mit Deiner gepatchten Version arbeiten möchte. Gibt es da einen Trick oder muss ich mir ein Telefon mit NFC kaufen? LG, Richard

**Comment by MasterPlexus** [comment]
> Erst einmal "Vielen Dank" für Deine Arbeit. :) Mein Pocophone F1 hat kein NFC und LibreLink verweigert aus dem Grund den Start. Eigentlich brauche ich NFC ja gar nicht, da ich per Bluetooth mit Deiner gepatchten Version arbeiten möchte. Gibt es da einen Trick oder muss ich mir ein Telefon mit NFC kaufen? LG, Richard

Nein, den gibt es nicht, denn über den ersten NfC-Scann wird der Sensor gepairt mit dem Handy. Ohne NfC kann der Sensor nicht verbunden werden.

No, this won't work, because the Sensor will be paired with the help of NfC-Scanning. Without NfC there is no way to pair the Sensor with the Phone.

**Comment by Rig22** [comment]
Ahh, schade, war mit dem Pocophone eigentlich bisher sehr zufrieden. Vielen Dank für Deine schnelle Antwort. :)

**Comment by Desacrel** [comment]
Hi, 
Ich stehe noch nicht vor dem Problem da er bei mir mit Fehlern nicht Patchen möchte.
Aber auf der Suche nach eine Lösung habe ich ein weiteres Problem enteckt. "kein NFC".
Nun zur Frage, gibt es eine andere Möglichkeit ein NFC Signal zu senden ?
Also nur zum anstoß von einem anderen Handy ? Oder über NFC-Tag ?

Lg Desa

**Comment by MasterPlexus** [comment]
@desacrel wie oben schon beschrieben, das wird nicht gehen. Die Optionen die du beschreibst würden auch so tiefgehende Kenntnisse des Betriebssystems und root benötigen, so daß das ganze nicht handhabbar wäre...

**Comment by acrobatba** [comment]
Hi
I never used phone or xdrip for scaning. Only Libre 2 with Libre 2 reader.
So, xdrip and phone scaning is new to me, and im sorry if my questions doesnt belong here. Im also very weak with android and phone things.
Im trying to help sister and her daughter.
I have couple questions :

- i think i did everything you said (i had 5 errors white space ),
and copied patched app on phone. I didnt get any errors when installing on phone.
That means patch is ok?
- i didnt have any previous Librelink app installed. Is that a problem?
- will this patched version work with already started sensor (just for test)?
- if something get wrong will i be able to use Libre reader? (sensor is so expensive for us)
 
 - i installed xdrip (my first time), what will i choose as source ?
 (now, without any sensor in use, i see options g4,g5... Libre with miaomiao...)
 - Librelink app is in use only to scan new sensor; rest is on xdrip?
 
 Im sorry if this is too many questions and if they dont belong here, but im doing everything this at my home, then i will go to test this to my sister house?
 
 Thanks

**Comment by Desacrel** [comment]
verdammt. na dann versuche ich es erst einmal mit einem alten

**Comment by elbingobonito** [comment]
>i think i did everything you said (i had 5 errors white space ),
>and copied patched app on phone. I didnt get any errors when installing on phone.
>That means patch is ok?

If you did everything according to the manual then yes. 

>i didnt have any previous Librelink app installed. Is that a problem?

According to #4 you need to have the original app from the playstore installed before. 

>will this patched version work with already started sensor (just for test)?

It should work to scan the sensor, but the bluetooth transmission shouldn't work. 

> if something get wrong will i be able to use Libre reader? (sensor is so expensive for us)

This of course depends on the "something", which might go wrong, but as an educated guess I would assume that you still will be able to use the libre 2 reader (just without the alarms). 

>  i installed xdrip (my first time), what will i choose as source ?

The source option is called "Libre 2 (patched)" and is available in the recent [nightly builds ](https://github.com/NightscoutFoundation/xDrip/releases)of xdrip+.

> Librelink app is in use only to scan new sensor; rest is on xdrip?

Yes.

**Comment by acrobatba** [comment]
elbingobonito , thank you for your time and answers.

**Comment by acrobatba** [comment]
What did I do wrong?

I scanned new sensor with Librelink.
I couldnt find source option "Libre 2 (patched)" .
I did bluetooth scan and it showed abbot......
Start collecting data. And nothing.
I tried to scan again and again but abbot.... never showed again :(

![20190723_212035](https://user-images.githubusercontent.com/53016467/61747971-6bb5e980-ad9f-11e9-9a4d-5144a932ab0f.jpg)
![20190723_212238](https://user-images.githubusercontent.com/53016467/61747972-6c4e8000-ad9f-11e9-9981-25bc86c0a9f0.jpg)
![20190723_213235](https://user-images.githubusercontent.com/53016467/61747976-6c4e8000-ad9f-11e9-82fb-45e0ac820f9c.jpg)
![20190723_213540](https://user-images.githubusercontent.com/53016467/61747977-6ce71680-ad9f-11e9-9bf5-9f0471c3a70d.jpg)

**Comment by TinoKossmann** [comment]
No "Libre 2 (patched App)" as source? Old Version of xDrip!

Get the latest "Nightly Build".

**Comment by acrobatba** [comment]
I downloaded file xDrip-plus-20190518-14a03a7. Can you send me link for Old Version of xDrip?

**Comment by TinoKossmann** [comment]
18.05.2019 ist too old. Here you can find the latest "nightly buids"(klick on "Assets" and download the APK):

[https://github.com/NightscoutFoundation/xDrip/releases](url)

**Comment by acrobatba** [comment]
Its working :)
Thanks everyone for help.
Can someone share his experience with Libre 2 patch?
Perhaps someone who used Libre 1 + miaomiao + xdrip. Is accuracy same, better, worst?
We noticed that signal was lost when girl went to another room, which never was case with Libre reader.

**Comment by TimMoritz** [comment]
>     if something get wrong will i be able to use Libre reader? (sensor is so expensive for us)
> This of course depends on the "something", which might go wrong, but as an educated guess I would assume that you still will be able to use the libre 2 reader (just without the alarms).

Sadly you can't connect the Libre 2 reader after activating the sensor with your phone - I tested it :(.
And I think if the sensor will be activated with the reader first and then connect the phone with (patched) Librelink that it does not get bluetooth connect to the sensor and xDrip+ gets no values, too. Has someone this tested this before?

Thanks

**Comment by Rig22** [comment]
Yes, no Bluetooth Connect If you activate the Libre 2 with the reader first. I made that mistake, when i started my first Libre 2.

**Comment by yoxud** [comment]
> You really have to install the original app from play store before you uninstall the app again and install the rebuilt version. 

I cannot install the original from the playstore because I am in Canada. Installing the patched version also leads to a "must install from Playstore" error. Is this solution only restricted to Germany?
